lib/fs.js

```diff
@@ -243,10 +247,18 @@ fs.readFile = function(path, options, callback_) {
     return;
 
   var context = new ReadFileContext(callback, encoding);
+  context.isUserFd = isFd(path); // file descriptor ownership
   var req = new FSReqWrap();
   req.context = context;
   req.oncomplete = readFileAfterOpen;
 
+  if (context.isUserFd) {
+    process.nextTick(function() {
+      req.oncomplete.call(req, null, path);
```

Comment:
It would not behave identically. This call is supposed to emulate the behavior of `binding.open()`. Omitting the context would lead to incompatibilities with the following code (ex: `readFileAfterOpen()`, where the context is extracted from `this`).